corehq/apps/api/resources/v0_4.py

```diff
@@ -120,6 +120,7 @@ def obj_get_list(self, bundle, domain, **kwargs):
 
         def wrapper(doc):
             if doc['doc_type'] in xform_doc_types:
+                doc.pop('user_id', None)  # needed when xform is stored in sql
                 return xform_doc_types[doc['doc_type']].wrap(doc)
```

Comment:
If I'm understanding the history of this correctly, I can imagine why this might be useful; You want to duck-type SQL instances as Couch instances, e.g. to use [over here][1]. Deserializing from JSON using an API SQL serializer might be a better approach, but it can be annoying because it tries to save the instance (which already exists). To avoid the save, you can do something like this (effectively a hacky "wrap" for SQL classes): 

```python
def get_instance_from_data(SerializerClass, data):
    """
    Return a deserialized instance from serialized data

    cf. https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/serializers.py#L71
    and https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/serializers.py#L882
    """
    # It does seem like you should be able to do this in one line. Django REST 
    # framework assumes that you always want to `save()`. This function avoids 
    # saving.
    ModelClass = SerializerClass.Meta.model
    serializer = SerializerClass(data=data)
    assert serializer.is_valid(), 'Unable to deserialize data while creating {}'.format(ModelClass)
    # ^^^ We need to call `is_valid()` to get to `validated_data` vvv
    return ModelClass(**serializer.validated_data)
```

The alternative is to wrap it as a Couch instance -- i.e. what I think @snopoke seems to have done here -- and which might turn out to be the better of the alternatives.


  [1]: https://github.com/dimagi/commcare-hq/blob/03c98ab17d6b8180ab638ffb7e0e520aec3b25e7/corehq/apps/api/es.py#L564-L564